subsys/mgmt/mcumgr/Kconfig

```
@@ -123,6 +123,12 @@ config FS_MGMT_PATH_SIZE
 	  and download commands.
 endif
 
+config MCUMGR_CMD_SHELL_MGMT
+	bool "Enable mcumgr handlers for shell management"
+	select SHELL
```

```suggestion
	depends on !SHELL_BACKEND_SERIA && SHELL
	select SHELL_BACKEND_DUMMY
```
^^ This makes impossible to misconfigure the feature you added.
You can also add same comments as in overlay-shell-mgmt.conf

---

Thanks for the feedback!

So the way I have designed it, you are allowed to use `SHELL_BACKEND_SERIAL` and I check for it in the mcumgr command handler. The only "issue" is that you can't get the shell output when using the serial backend. The command is still executed, and you still get the return code, and the shell output is still sent through the serial output, but mcumgr can't get it. So basically I don't enforce the dummy backend, it is just a more complete implementation if you are using it.

If that is confusing, I am okay with just enforcing the dummy backend and I would change the implementation in the mcumgr PR.

However you bring up a good point because it is going to be impossible to use `CONFIG_MCUMGR_SMP_SHELL` with this, so I should have `depends on !CONFIG_MCUMGR_SMP_SHELL` for sure.

---

So `imply SHELL_BACKEND_DUMMY` might be used instead (see https://docs.zephyrproject.org/latest/guides/kconfig/tips.html?highlight=imply#the-imply-statement).

---

@de-nordic Make it sense to you that the shell response arrives by another channel? I afraid that this might be notices as a bug by the user.

---

While returning result code may be useful even if no command output gets received, e.g. while toggling some parameters, reporting output to other channel may break something on user side.

---

That makes sense. So probably the best solution is to just enforce the dummy backend. I will change this in the implementation and then make the changes suggested here. Thanks!

---

The mcumgr stat command seems to be able to pack, as cbor, output on device and read it on host via mcumgr. Wouldn't it be possible to do the same here?

---

@de-nordic @nvlsianpu  Thanks again for the feedback. I went ahead and removed the support for serial backend from mcumgr, so as to not introduce confusing behavior where it isn't needed. I then changed the implementation in this repo now to reflect that and used the suggestions provided here to prevent misconfiguration. 

Dummy backend is now enforced.